### PR TITLE
Domh/fixing issues from old prs

### PR DESCRIPTION
**Pull request by DomHumphrey-Microsoft**

### CODE REVIEW

File: imgui.cpp [hunk 1]
```diff
@@ -851,12 +983,13 @@ CODE
 #endif
 
 #include "imgui.h"
+#pragma optimize("", off)
```
Comment (spierceMicrosoft) [inline imgui.cpp]: Revert

File: imgui.h [hunk 2]
```diff
@@ -13,19 +13,22 @@ Index of this file:
...
+// Draw List API (ImDrawCallback, ImDrawCmd, ImDrawIdx, ImDrawVert, ImDrawChannel, ImDrawListFlags, ImDrawList, ImDrawData)
+// Font API (ImFontConfig, ImFontGlyph, ImFontGlyphRangesBuilder, ImFontAtlasFlags, ImFontAtlas, ImFont)
+// Platform interface for multi-viewport support (ImGuiPlatformMonitor, ImGuiPlatformIO, ImGuiViewport)
 
 */
 
 #pragma once
-
-// Configuration file (edit imconfig.h or define IMGUI_USER_CONFIG to your own filename)
+//#include "Core/include/Core/Debug/Log.h"
```
Comment (spierceMicrosoft) [inline imgui.h]: Imgui had better not know about the Core project, or any first party libraries for that matter.

**Review by spierceMicrosoft — CHANGES_REQUESTED** [review]
Doing gigantic changes to these third party repositories makes them very difficult to update and maintain. It's one thing to fork for a couple bug fixes and possibly report those fixes or contribute back so you don't need to fork anymore. But these big additions just shoot yourself in the foot.

This is not how you sync a fork. This doesn't preserve history from the original repository. I can't tell what is your code and what is external code. You loose all history doing it this way.

Fixes like warning disables should be done via the build system, not by editing third party code.

Also this is a **PUBLIC** fork. Anyone on the internet can see it. Be very careful what you put into it.

File: imgui.cpp (hunk 1 above)
Comment (DomHumphrey-Microsoft) [reply imgui.cpp]: sorry - missed adding the file that fixes this before creating the PR.  Fixed

**Comment by DomHumphrey-Microsoft** [comment]
If you'd like this to happen differently, please do come by my desk and teach me.  I'm completely new to sub-modules and such, so if I'm doing it wrong, I'd love to know the best practices.

File: imgui.h (hunk 2 above)
Comment (DomHumphrey-Microsoft) [reply imgui.h]: fixed

**Comment by spierceMicrosoft** [comment]
Generally you need to setup a new upstream remote that points to the repository you forked from. Your current remote is origin and points to this repository. You just need to use `git remote add` to create an upstream if it doesn't exist.

I haven't personally updated any forks in the Mojang org, so I'm not sure if you need https syntax like submodules. 

But generally it would look like `git remote add upstream git@github.com:ocornut/imgui.git`

You can think of that like a second `origin` that just points to a slightly different location.

To update the fork, it should just be roughly the normal workflow but using the upstream instead.
`git fetch upstream`
`git rebase -i upstream/master`.

**Pull request closed without merging**